internal/tasks/templateversion_populate.go

```diff
@@ -219,6 +221,31 @@ func (t *TemplateVersionPopulateLogic) handleInlineConfig(configItem *api.Device
 	return nil
 }
 
+func (t *TemplateVersionPopulateLogic) handleHttpConfig(configItem *api.DeviceSpec_Config_Item) error {
+	httpSpec, err := configItem.AsHttpConfigProviderSpec()
+	if err != nil {
+		return fmt.Errorf("failed getting config item as HttpConfigProviderSpec: %w", err)
+	}
+
+	repo, err := t.store.Repository().GetInternal(context.Background(), t.resourceRef.OrgID, httpSpec.HttpRef.Repository)
+	if err != nil {
+		return fmt.Errorf("failed fetching specified Repository definition %s/%s: %w", t.resourceRef.OrgID, httpSpec.HttpRef.Repository, err)
+	}
+
+	if repo.Spec == nil {
+		return fmt.Errorf("empty Repository definition %s/%s: %w", t.resourceRef.OrgID, httpSpec.HttpRef.Repository, err)
+	}
+
+	newConfig := &api.TemplateVersionStatus_Config_Item{}
```

Comment:
We should "freeze" the configuration here.  The best way to do that right now is to fetch the contents and add it as an inline config.

Reply:
Thanks for the suggestion. Let me do that!